beacon-chain/p2p/handshake.go

```diff
@@ -22,7 +22,57 @@ const (
 )
 
 func peerMultiaddrString(conn network.Conn) string {
-	return fmt.Sprintf("%s/p2p/%s", conn.RemoteMultiaddr().String(), conn.RemotePeer().String())
+	remoteMultiaddr := conn.RemoteMultiaddr().String()
+	remotePeerID := conn.RemotePeer().String()
+	return fmt.Sprintf("%s/p2p/%s", remoteMultiaddr, remotePeerID)
+}
+
+func (s *Service) connectToPeer(conn network.Conn) {
+	s.peers.SetConnectionState(conn.RemotePeer(), peers.PeerConnected)
+	// Go through the handshake process.
+	log.WithFields(logrus.Fields{
+		"direction":   conn.Stat().Direction.String(),
+		"multiAddr":   peerMultiaddrString(conn),
+		"activePeers": len(s.peers.Active()),
+	}).Debug("Initiate peer connection")
+}
+
+func (s *Service) disconnectFromPeer(
+	conn network.Conn,
+	goodByeFunc func(ctx context.Context, id peer.ID) error,
+	badPeerErr error,
+) {
+	// Get the remote peer ID.
+	remotePeerID := conn.RemotePeer()
+
+	// Get the direction of the connection.
+	direction := conn.Stat().Direction.String()
+
+	// Get the remote peer multiaddr.
+	remotePeerMultiAddr := peerMultiaddrString(conn)
+
+	// Set the peer to disconnecting state.
+	s.peers.SetConnectionState(remotePeerID, peers.PeerDisconnecting)
+
+	// Only attempt a goodbye if we are still connected to the peer.
+	if s.host.Network().Connectedness(remotePeerID) == network.Connected {
+		if err := goodByeFunc(context.TODO(), remotePeerID); err != nil {
+			log.WithError(err).Error("Unable to disconnect from peer")
+		}
+	}
+
+	// Get the remaining active peers.
+	activePeerCount := len(s.peers.Active())
+	log.
+		WithError(badPeerErr).
+		WithFields(logrus.Fields{
+			"multiaddr":            remotePeerMultiAddr,
+			"direction":            direction,
+			"remainingActivePeers": activePeerCount,
+		}).
+		Debug("Initiate peer disconnection")
```

Comment:
This will be noisy on mainnet, do we really need to log this ?

Reply:
At least for devnet, it's important to know why a peer is disconnected.
Is it from our initiative? Is it from the peer initiative?
If it is from our initiative, what was the reason? Bad gossip score? Too many bad response? Other?

If it is too noisy on mainnet, or even on testnet, I can only print this log only if we are not on mainnet/testnet.

Reply:
We could leave this in the `peerDAS` branch if it helps with debugging but not bring it down to `develop`